### PR TITLE
Remove unstable AST testing

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

https://github.com/babel/babel/issues/12915 has been fixed so we can remove unstable AST testing.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
Tests are failing via babel/babel#12939 for `json-stringify` parser.

File: tests_config/run_spec.js [hunk 1]
```diff
@@ -47,19 +47,18 @@ const unstableTests = new Map(
...
+      (options) => options.parser === "json-stringify",
     ],
     [
-      "flow/function-parentheses/test.js",
-      (options) => options.parser.startsWith("babel"),
+      "json/json/positive-number.json",
+      (options) => options.parser === "json-stringify",
     ],
+    ["json/json/json5.json", (options) => options.parser === "json-stringify"],
+    ["json/json/array.json", (options) => options.parser === "json-stringify"],
```
Comment (sosukesuzuki) [inline tests_config/run_spec.js:61]: In now `parserExpression` returns AST includes `tokens` so may break AST idempotency for `json-stringify` parser.

Comment (thorn0) [reply tests_config/run_spec.js:61]: I see `tokens` in `ignoredProperties` in `clean.js`. Why does that property still affect idempotence tests?

File: yarn.lock [hunk 2]
```diff
@@ -283,7 +283,12 @@
...
     js-tokens "^4.0.0"
 
-"@babel/parser@7.13.4", "@babel/parser@^7.1.0", "@babel/parser@^7.12.13", "@babel/parser@^7.13.0", "@babel/parser@^7.13.4":
+"@babel/parser@7.13.9":
+  version "7.13.9"
+  resolved "https://registry.yarnpkg.com/@babel/parser/-/parser-7.13.9.tgz#ca34cb95e1c2dd126863a84465ae8ef66114be99"
+  integrity sha512-nEUfRiARCcaVo3ny3ZQjURjHQZUo/JkEw7rLlSZy/psWGnvwXFtPcr6jb7Yb41DVW5LTe6KRq9LGleRNsg1Frw==
+
+"@babel/parser@^7.1.0", "@babel/parser@^7.12.13", "@babel/parser@^7.13.0", "@babel/parser@^7.13.4":
   version "7.13.4"
```
Comment (thorn0) [inline yarn.lock:292]: This needs deduplication.

File: tests_config/run_spec.js (hunk 1 above)
Comment (sosukesuzuki) [reply tests_config/run_spec.js:61]: Probably `ignoreProperties` are ignored by `messageAST`, are not by idempotence tests.

https://github.com/prettier/prettier/blob/main/src/main/massage-ast.js#L13-L18

Comment (fisker) [reply tests_config/run_spec.js:61]: It should work, I'll take a look.

Comment (fisker) [reply tests_config/run_spec.js]: `tokens` was not ignored in `estree-json` printer.

**Review by thorn0 — APPROVED** [review]

**Pull request merged**